### PR TITLE
Move conversation GC to data

### PR DESCRIPTION
**Pull request by elverkilde**
Minor refactoring of the conversation "garbage collection".

### CODE REVIEW

File: services/galley/src/Galley/Data.hs [hunk 1]
```diff
@@ -295,7 +295,18 @@ conversation :: (MonadBaseControl IO m, MonadClient m, Forall (Pure m))
...
+    mbConv <- toConv conv <$> members conv <*> wait cdata
+    return mbConv >>= conversationGC
+
+
+{- "Garbage collect" the conversation, i.e. the conversation may be
+   marked as deleted, in which case we delete it and return Nothing -}
+conversationGC :: MonadClient m => (Maybe Conversation) -> m (Maybe Conversation)
+conversationGC conv = case join (convDeleted <$> conv) of
+    (Just True) -> do
+        _ <- sequence $ deleteConversation . convId <$> conv
```
Comment (tiago-loureiro) [inline services/galley/src/Galley/Data.hs]: `sequence_ $ deleteConversation . convId <$> conv` ?

**Comment by elverkilde** [comment]
@tiago-loureiro @jschaul rebased

**Comment by tiago-loureiro** [comment]
> @tiago-loureiro @jschaul rebased

But not pushed?

**Comment by jschaul** [comment]
> But not pushed?

I suppose it was, but I merged my PR in, so github states "out-of-date" again.

**Pull request merged**